### ISSUE TITLE
feature request: specify the widths for pipe tables

### ISSUE THREAD

**Issue by ndaniel** [jgm/pandoc#2486]
This issue is derived from [issue #2471](https://github.com/jgm/pandoc/issues/2471#issuecomment-150205451).

Currently, pandoc does not have the ability to extract or use the widths of columns for pipe tables.
It would be great, if the table width and the columns' width could be specified for pipe tables by using **only and only** this line:

```
|:---|----|--------|:----------:|
```

from the pipe table:
```
| a | b | c | d |
|:---|----|--------|:----------:|
| 1 | 2 | 3 | 44444444 |
| 101 010 101 | 5 | 6 | 7 |
```
where first  column and second column have the same width, the third column is twice wider than first column, and fourth column is three times wider than first column).

If 
```
|:---|----|--------|:----------:|
```
is longer than the page width then the table is allowed to overflow the page width.

Pros:
  * the computation of widths is much the same way is done for multiline tables currently
  * no changes required to the markdown syntax
  * no changes required to pipe tables syntax
  * no latex/html special packages/hacks are needed
  * no more pipe table overflowing the page width in PDF

Cons:
....

**Comment by jgm** [comment]
One possible issue with this is that, for tables with small cells, the current behavior is usually better -- LaTeX will precisely fit the table cells.  Perhaps the logic could be as follows:  If all the rows of the table are < the column width (as specified by `--columns`, default 80), produce a "simple" table with no column width information.  If any of the rows are longer than column width, compute relative widths based on the separator lines, as suggested above.

**Comment by ndaniel** [comment]
I agree that in some situations the present behavior of pipe tables is desirable, like for example tables with small cells.

Therefore, a suggestion might be as follows.
If **all** the widths of **all** columns, which are specified in this line 
```
|:---|----|--------|:----------:|
```
are less or equal to 3 characters in markdown (this could be even 2?) then the pipe tables will behave as they behave currently (that is no widths are used).

For example, this
```
|:--|--|:-:|-|
```
would indicate that no widths info for the pipe table will be used (because all the columns widths are less or equal to 3 characters in markdown source) and the table will behave like today. Anyway for such small numbers (like  1, 2, and 3)  is really difficult to convey for the user the info regarding the columns widths and table width.

**But** for example, this
```
|:--|---|--------|:----------:|
```
would indicate that widths info for the pipe table will be used (because there is at least one column which has the width strictly larger than 3 characters).

The advantage of this is that no changes are required to the markdown syntax and the **today's** behavior of pipe table can still be preserved in the future.

**Comment by jgm** [comment]
I think I still prefer the suggestion I made.
After all, many people using tables with small cells
will use bars with more than three characters, just to
make them look nice in the source:

```
|fruit | quantity |
|------|---------:|
| apple|        33|

```

On your proposal, tables like this will render differently
than they did before, and the change might not be welcome.
So I'm inclined to be a little more conservative about
kicking in cell widths.

On the other hand, a table with, say, 50 3-character cells
will probably still need to wrap cells, and will need
widths.  That's why it makes more sense to me to look at
the total width of the separator lines, rather than
individual widths.


+++ Daniel Nicorici [Oct 28 15 00:47 ]:

**Comment by ndaniel** [comment]
You have a good point! :-)

**Comment by jgm** [comment]
Closed by fix to #2471.  Let me know if you try this and find it unsatisfactory.
